Northstar.Custom/mod/scripts/vscripts/sh_northstar_safe_io.gnut

```diff
@@ -0,0 +1,35 @@
+globalize_all_functions
+
+table< int, void functionref( string ) > pendingCallbacks
+table< int, void functionref( table ) > pendingJSONCallbacks
+
+void function NSLoadFile( string file, void functionref( string ) onSuccess )
+{
+    int handle = NS_InternalLoadFile( file )
+
+    pendingCallbacks[handle] <- onSuccess
+}
+
+void function NSLoadJSONFile( string file, void functionref( table ) onSuccess )
+{
+    int handle = NS_InternalLoadFile( file )
+
+    pendingJSONCallbacks[handle] <- onSuccess
+}
+
+void function NSHandleLoadResult( int handle, string result )
+{
+    if (handle in pendingCallbacks)
+    {
+        pendingCallbacks[handle](result)
+        delete pendingCallbacks[handle]
+        return
+    }
+    if (handle in pendingJSONCallbacks)
+    {
+        pendingJSONCallbacks[handle](DecodeJSON(result))
+        delete pendingJSONCallbacks[handle]
+        return
+    }
+    unreachable
```

Comment:
This is reachable.
```suggestion
    throw "Invalid IO callback handle"
```

Reply:
Proof that it is reachable?

Reply:
If you call this function with a handle that is neither in `pendingCallbacks` nor in `pendingJSONCallbacks`

Reply:
You are not supposed to call this function from script at all.

This function is only called from native.

Reply:
`unreachable` is supposed to be a hint for the compiler that this function has already returned. If there is any way to reach the statement at all, `throw` should be used as it behaves the same but gives more context to what happens

Reply:
Yes, because the function should already return - it's either a JSON callback that needs to be done, or a normal callback. If it's neither (which should never happen because Safe I/O should never call an invalid handle) then it's a problem.

There is no context needed.

Reply:
This isn't really the intended use for `unreachable`, `unreachable` is only meant for when the dumb squirrel compiler cant tell that every code path is returning a value already (although respawn don't always follow this)

This is not the case here, and a `throw` with a simple error message like `"I/O handle "+ handle + "has no callbacks, is NSHandleLoadResult being called improperly?"` would be better imo. Having "code reached unreachable" or whatever the error is is less useful than a properly thrown error message.